src/components/nano/ModalRegisterNanoContract.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useEffect, useRef, useState } from 'react';
+import { t } from 'ttag';
+import $ from 'jquery';
+import { get } from 'lodash';
+import { useDispatch, useSelector } from 'react-redux';
+import { cleanNanoContractRegisterMetadata, registerNanoContract } from '../../actions/index';
+import { getGlobalWallet } from "../../modules/wallet";
+import { NANOCONTRACT_REGISTER_STATUS } from '../../sagas/nanoContract';
+import { colors } from '../../constants';
+import hathorLib from '@hathor/wallet-lib';
+import ReactLoading from 'react-loading';
+
+/**
+ * Component that shows a modal to register a Nano Contract
+ *
+ * @memberof Components
+ */
+function ModalRegisterNanoContract({ onClose }) {
+  const dispatch = useDispatch();
+  const wallet = getGlobalWallet();
+
+  const registerNCModalID = 'registerNCModal';
+
+  const [ncId, setNcId] = useState('');
+  const ncRegisterMetadata = useSelector(state => state.nanoContractsRegisterMetadata);
+  const ncRegisterStatus = get(ncRegisterMetadata, 'status', null);
+
+  const formRegisterNCRef = useRef(null);
+
+  const idRef = useRef(null);
+
+  useEffect(() => {
+    $(`#${registerNCModalID}`).modal('show');
+    $(`#${registerNCModalID}`).on('hidden.bs.modal', (e) => {
+      dispatch(cleanNanoContractRegisterMetadata());
+      idRef.current.value = '';
+      // We always need to call on close when using global context modal
+      onClose(registerNCModalID);
```

Comment:
question: Did you test this? I'm afraid this method requires the jQuery selector `#${registerNCModalID}`, not only the identifier.

It's actually an improvement point for the Global Modal, but currently I think this would not work properly.

Reply:
You are correct, fixed https://github.com/HathorNetwork/hathor-wallet/pull/529/commits/9c950018367442d1c13b45552ee6a685bb256b6b